### PR TITLE
[MM-13704] Updated CLI "deleter user" to also permanently delete GroupMembers associated to user

### PR DESCRIPTION
**Pull request by hahmadia**
#### Summary
Updated code so that when you run `mattermost delete user` it will also permanently delete the user from the GroupMembers table in the database

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-13704

### CODE REVIEW

File: store/sqlstore/group_store.go [hunk 1]
```diff
@@ -341,6 +341,13 @@ func (s *SqlGroupStore) DeleteMember(groupID string, userID string) (*model.Grou
 	return retrievedMember, nil
 }
 
+func (s *SqlGroupStore) PermanentDeleteMembersByUser(userId string) *model.AppError {
+	if _, err := s.GetMaster().Exec("DELETE FROM GroupMembers WHERE UserId = :UserId", map[string]interface{}{"UserId": userId}); err != nil {
+		return model.NewAppError("SqlGroupStore.RemoveMember", "store.sql_group.permanent_delete_members_by_user.app_error", nil, "user_id="+userId+", "+err.Error(), http.StatusInternalServerError)
```
Comment (Willyfrog) [inline store/sqlstore/group_store.go]: I think user_id should be on the params argument, leaving the details only with the original error.

Comment (hahmadia) [reply store/sqlstore/group_store.go]: Done! Thanks 👍

File: i18n/en.json [hunk 2]
```diff
@@ -6126,6 +6126,10 @@
     "id": "store.sql_group.no_rows_changed",
     "translation": "no rows changed"
   },
+  {
+    "id": "store.sql_group.permanent_delete_members_by_user.app_error",
+    "translation": "Unable to remove the group member with UserID \"{{.UserId}}\""
```
Comment (mkraft) [inline i18n/en.json]:
```suggestion
    "translation": "Unable to remove the group members with UserID \"{{.UserId}}\""
```

File: store/sqlstore/group_store.go [hunk 3]
```diff
@@ -341,6 +341,13 @@ func (s *SqlGroupStore) DeleteMember(groupID string, userID string) (*model.Grou
 	return retrievedMember, nil
 }
 
+func (s *SqlGroupStore) PermanentDeleteMembersByUser(userId string) *model.AppError {
+	if _, err := s.GetMaster().Exec("DELETE FROM GroupMembers WHERE UserId = :UserId", map[string]interface{}{"UserId": userId}); err != nil {
+		return model.NewAppError("SqlGroupStore.RemoveMember", "store.sql_group.permanent_delete_members_by_user.app_error", map[string]interface{}{"UserId": userId}, "", http.StatusInternalServerError)
```
Comment (mkraft) [inline store/sqlstore/group_store.go]:
```suggestion
		return model.NewAppError("SqlGroupStore.PermanentDeleteMembersByUser", "store.sql_group.permanent_delete_members_by_user.app_error", map[string]interface{}{"UserId": userId}, "", http.StatusInternalServerError)
```

**Comment by mkraft** [comment]
@hahmadia This requires tests, please.

**Comment by hahmadia** [comment]
> @hahmadia This requires tests, please.

Addressed! Thanks for pointing that out.

File: store/storetest/group_store.go [hunk 4]
```diff
@@ -658,6 +659,42 @@ func testGroupDeleteMember(t *testing.T, ss store.Store) {
...
+
+	// Create members
+	for _, group := range groups {
+		_, err = ss.Group().UpsertMember(group.Id, user.Id)
+		require.Nil(t, err)
+	}
+
+	// Happy path
+	err = ss.Group().PermanentDeleteMembersByUser(user.Id)
+	require.Nil(t, err)
```
Comment (mkraft) [inline store/storetest/group_store.go]: Maybe worth adding a test where you pass in and empty string and/or an invalid id to test that an error is returned.
